prow/github/report/report.go

```diff
@@ -189,6 +200,16 @@ func Report(ghc GithubClient, reportTemplate *template.Template, pj prowapi.Prow
 	return nil
 }
 
+// commitIsLatest returns true if a given commit SHA is found
+// to be the last one in a slice of RepositoryCommits.
+func commitIsLatest(sha string, commits []github.RepositoryCommit) bool {
+	total := len(commits)
+	if total == 0 {
+		return false
+	}
+	return sha == commits[total-1].SHA
```

Comment:
IIRC there is some weirdness with ordering in this API -- @cjwagner are we certain we get the commits in logical order here?

Reply:
yes, also interested in this.

Reply:
/assign @cjwagner

Reply:
I don't know how the v3 API is ordered, the v4 API is the one that orders commits by author date. Since the v3 API doesn't specify an ordering I don't think we should rely on it. https://developer.github.com/v3/pulls/#list-commits-on-a-pull-request

What about fetching the PullRequest itself and comparing against the `.Head.SHA` field? That also has the benefit of not being a paginated endpoint so it will only ever cost 1 API token.